image-processing/UI.py

```diff
@@ -1,484 +1,202 @@
+        if grouping in ('high', 'low'):
+            return grouping
+        
+        # elif grouping.lower() == "back":
+        #     self.input_sample()
 
-                            if view_plot == 'yes':
-                                print('Result for Sample: {}'.format(paths_saved[scan].stem))
+        else:
+            print("Response not recognised, please respond with either 'low' or 'high'.\n")
+            self.input_grouping()
+    
+    def input_view_plot(self):
+        view_plot = input("Would you like to view the output plot from the analysis software? ").lower()
 
-                                if sampleType == 1:
-                                    finder.severity_carbon(4, scans_saved[scan][1])
+        if view_plot in ('yes', 'no'):
+            return view_plot
+        
+        # elif view_plot.lower() == "back":
+        #     self.input_grouping()
 
-                                if sampleType == 2:
-                                    finder.severity_metal_coated(6, scans_saved[scan][1])
+        else:
+            self.yesno_error()
+            self.input_view_plot()
 
-                                name = str(paths_saved[scan].stem)
-                                finder.plot_nice(name = name)
+    def input_baseline(self):
+        baseline = input('\nWhat severity out of 10 is required to fail a sample? \nRecommendations: 4 for Carbon Veil, 6 for Metal-Coated Carbon Veil \nEnter Value: ')    #got rid of divide by 2 for now
+        
+        if baseline.isnumeric():
+            return int(baseline)
+        
+        # elif baseline.lower() == "back":
+        #     self.input_view_plot()
+        
+        else:
+            print("Invalid input. Please ensure a positive number is entered.\n")
+            self.input_baseline()
 
-                            elif view_plot == 'no':
-                                print('Result for Sample:  {}'.format(paths_saved[scan].stem))
+    def input_set_parameters(self):
+        set_parameters = input("Would you like to use preset values or set your own parameters? ('yes' for presets, 'no' for custom parameters) ").lower()
 
-                                if sampleType == 1:
-                                    finder.severity_carbon(4, scans_saved[scan][1])
+        if set_parameters in ('yes', 'no'):
+            return set_parameters
+        
+        # elif set_parameters.lower() == "back":
+        #     self.input_baseline()
 
-                                if sampleType == 2:
-                                    finder.severity_metal_coated(6, scans_saved[scan][1])
+        else:
+            self.yesno_error()
+            self.input_set_parameters()
 
+    def input_sigma(self):
+        sigma = input("Set sigma value of Gaussian blur to determine severity of image blurring. Recommended value is 1: ")     #and can be... what? what type of number can they input??
```

Comment:
sigma doesn't have to be an integer no:)

Reply:
to change (MC)